### PR TITLE
[Data] Update iteration release tests

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

The current iterations are too small-scale to capture real-world issues. To address this, this PR makes the iteration release tests multi-node and larger-scale.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_data_tests.yaml [hunk 1]
```diff
@@ -60,7 +61,9 @@
 - name: write_parquet
   run:
     timeout: 3600
-    script: python read_and_consume_benchmark.py s3://ray-benchmark-data/parquet/1TiB --format parquet --write
+    script: > 
+      python read_and_consume_benchmark.py 
+      s3://ray-benchmark-data/tpch/parquet/sf1000/lineitem --format parquet --write
```
Comment (bveeramani) [inline release/release_data_tests.yaml:66]: Drive-by change to update the read, write, and count datasets to be more canonical (e.g., TPCH rather than my random synthetic dataset)

File: release/nightly_tests/dataset/read_and_consume_benchmark.py [hunk 2]
```diff
@@ -75,5 +109,6 @@ def consume_fn(ds):
 
 
 if __name__ == "__main__":
+    ray.init()
```
Comment (omatthew98) [inline release/nightly_tests/dataset/read_and_consume_benchmark.py:112]: OOC why did we need to add this here?

**Review by omatthew98 — APPROVED** [review]
one q but lgtm

File: release/nightly_tests/dataset/read_and_consume_benchmark.py (hunk 2 above)
Comment (bveeramani) [reply release/nightly_tests/dataset/read_and_consume_benchmark.py:112]: So that Ray initialization time isn't included in the benchmark time

Comment (omatthew98) [reply release/nightly_tests/dataset/read_and_consume_benchmark.py:112]: Ah makes sense, thanks!
